nixos/modules/system/boot/systemd-nspawn.nix

```diff
@@ -44,6 +44,23 @@ let
 
   instanceOptions = {
     options = sharedOptions // {
+      restartOnChange = mkOption {
+        default = true;
```

Comment:
Would it be better to make this `false` by default as well?

Reply:
We actively flipped the default for this, as part of the network initialization unfortunately happens inside a shellscript outside the container, and doesn't get applied if we just apply the config inside the container.